### PR TITLE
Make the BaseTestHarness's start method correctly wait until all handles are ready before finishing.

### PR DESCRIPTION
**Pull request by jasonwyatt**
This will be important with the upcoming scheduler usage stuff.

### CODE REVIEW

File: java/arcs/sdk/testing/BaseTestHarness.kt [hunk 1]
```diff
@@ -110,14 +120,36 @@ open class BaseTestHarness<P : Particle>(
...
+                //    }
+                //}
+                // TODO: get rid of this version when onReady is non-suspending
+                val job = Job()
+                handle.onReady {
+                    val ready = readySoFar.incrementAndGet()
+                    job.complete()
+                    launch { particle.onHandleSync(handle, ready == handles.size) }
+                }
+                withTimeout(1000) { job.join() }
```
Comment (jasonwyatt) [inline java/arcs/sdk/testing/BaseTestHarness.kt:149]: These todos are resolved in my main PR.

File: java/arcs/sdk/testing/BaseTestHarness.kt [hunk 2]
```diff
@@ -110,14 +120,36 @@ open class BaseTestHarness<P : Particle>(
...
+                //    }
+                // }
+                // TODO: get rid of this version when onReady is non-suspending
+                val job = Job()
+                handle.onReady {
+                    val ready = readySoFar.incrementAndGet()
+                    job.complete()
+                    launch { particle.onHandleSync(handle, ready == handles.size) }
+                }
+                withTimeout(1000) { job.join() }
```
Comment (ianull) [inline java/arcs/sdk/testing/BaseTestHarness.kt:149]: How the 1000 was determined? If a timeout really happens and an exception is thrown, the enclosing coroutineScope will be cancelled completely. Is it intended?

Comment (jasonwyatt) [reply java/arcs/sdk/testing/BaseTestHarness.kt:149]: Yeah, if a timeout happens here, the whole world has broken - so it makes sense to fail the test.

Comment (jasonwyatt) [reply java/arcs/sdk/testing/BaseTestHarness.kt:149]: The 1000 was just a ballpark for now - we'll get rid of it alltogether once I get the final stuff submitted and these todos resolved.

Comment (ianull) [reply java/arcs/sdk/testing/BaseTestHarness.kt:149]: SGTM.

**Pull request merged**